### DISCUSSION TITLE
Icon and text overlapping in symbol layer

### DISCUSSION THREAD

**Discussion by xinyuluo** [maplibre/maplibre-gl-js#1097] [Q&A] [answered]
Hi! I have a symbol layer which contains some icons and their corresponding text. I set `'icon-allow-overlap': true` because I'd like to show all icons and I'm ok with them overlapping. But I'd like that texts do not conflict with any nearby icons when zooming in/out (I'm okay if not all texts are displayed). However, it seems that text would overlap with *other* icons (i.e. not their corresponding icons) when I zoom out. I've read through the documentation and tried options like `'text-optional': true`, `'text-allow-overlap': false` etc. it still didnt solve my issue. I wonder what am I missing in this case? Thank you!

### Illustration
<img width="264" alt="image" src="https://user-images.githubusercontent.com/25223376/158301689-1d072428-5c9f-4429-b6ef-c7925fac4d31.png">

**Comment by xabbu42** [comment, marked as answer]
This is a bug and I'm working (unfortunately with limited time at the moment) at a fix in #985.

**Reply by xinyuluo** [reply]
I see, thank you!

**Comment by drwestco** [comment]
One workaround currently available is to have the icons and text in two separate layers.
